frontend/src/components/builder/events/events-selected-action.tsx

```diff
@@ -233,121 +255,268 @@ export function AgentOutputEvent({
     </div>
   )
 }
+
 export function SystemPromptPartComponent({
   part,
+  defaultExpanded = false,
 }: {
   part: SystemPromptPart
+  defaultExpanded?: boolean
 }) {
+  const [isExpanded, setIsExpanded] = React.useState(defaultExpanded)
+  const content =
+    typeof part.content === "string"
+      ? part.content
+      : JSON.stringify(part.content, null, 2)
+
+  const TRUNCATE_LIMIT = 200
+  const shouldTruncate = content.length > TRUNCATE_LIMIT
+
+  // For collapsed view: normalize whitespace and truncate
+  const normalizedContent = content.replace(/\s+/g, " ").trim()
+  const displayContent = isExpanded
+    ? content
+    : shouldTruncate
+      ? normalizedContent.substring(0, TRUNCATE_LIMIT) + "..."
+      : normalizedContent
+
   return (
-    <Card className="rounded-lg border-[0.5px] bg-muted/40 p-3 text-xs leading-normal shadow-sm">
+    <Card
+      className="cursor-pointer rounded-lg border-[0.5px] bg-muted/40 p-3 text-xs leading-normal shadow-sm hover:bg-muted/50"
+      onClick={() => setIsExpanded(!isExpanded)}
+    >
       <div className="flex items-start gap-2">
         <CaseUserAvatar user={SYSTEM_USER} size="sm" />
-        <div className="overflow-x-auto whitespace-pre-wrap break-words">
-          {typeof part.content === "string"
-            ? part.content
-            : JSON.stringify(part.content, null, 2)}
+        <div
+          className={`flex-1 overflow-x-auto break-words ${isExpanded ? "whitespace-pre-wrap" : "whitespace-nowrap"}`}
```

Comment:
```suggestion
          className={cn("flex-1 overflow-x-auto break-words", isExpanded ? "whitespace-pre-wrap" : "whitespace-nowrap")}
```
import cn from utils